cmd/influxd/server_integration_test.go

```diff
@@ -592,16 +592,55 @@ func runTestsData(t *testing.T, testName string, nodes Cluster, database, retent
+			                                                                     {"name": "cpu", "timestamp": "2010-02-28T01:03:37.703820946Z", "tags": {"host": "server02"}, "fields": {"value": 200}},
+			                                                                     {"name": "cpu", "timestamp": "2012-02-28T01:03:38.703820946Z", "tags": {"host": "server03"}, "fields": {"value": 300}}]}`,
 			query:    `SELECT value FROM "%DB%"."%RP%".cpu WHERE host = 'server01'`,
 			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","value"],"values":[["2015-02-28T01:03:36.703820946Z",100]]}]}]}`,
 		},
 		{
+			name:     "WHERE tags SELECT single field (2 EQ tags)",
+			query:    `SELECT value FROM "%DB%"."%RP%".cpu WHERE host = 'server01' AND region = 'us-west'`,
+			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","value"],"values":[["2015-02-28T01:03:36.703820946Z",100]]}]}]}`,
+		},
+		{
+			name:     "WHERE tags SELECT single field (1 EQ and 1 NEQ tag)",
+			query:    `SELECT value FROM "%DB%"."%RP%".cpu WHERE host = 'server01' AND region != 'us-west'`,
+			expected: `{"results":[{}]}`,
+		},
+		{
 			name:     "WHERE tags SELECT single field (EQ tag value2)",
 			query:    `SELECT value FROM "%DB%"."%RP%".cpu WHERE host = 'server02'`,
 			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","value"],"values":[["2010-02-28T01:03:37.703820946Z",200]]}]}]}`,
 		},
+		{
+			name:     "WHERE tags SELECT single field (NEQ tag value1)",
+			query:    `SELECT value FROM "%DB%"."%RP%".cpu WHERE host != 'server01'`,
+			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","value"],"values":[["2010-02-28T01:03:37.703820946Z",200],["2012-02-28T01:03:38.703820946Z",300]]}]}]}`,
+		},
+		{
+			name:     "WHERE tags SELECT single field (NEQ tag value1 AND NEQ tag value2)",
+			query:    `SELECT value FROM "%DB%"."%RP%".cpu WHERE host != 'server01' AND host != 'server02'`,
+			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","value"],"values":[["2012-02-28T01:03:38.703820946Z",300]]}]}]}`,
+		},
+		{
+			name:     "WHERE tags SELECT single field (NEQ tag value1 OR NEQ tag value2)",
+			query:    `SELECT value FROM "%DB%"."%RP%".cpu WHERE host != 'server01' OR host != 'server02'`, // Yes, this is always true, but that's the point.
+			expected: `{"results":[{"series":[{"name":"cpu","columns":["time","value"],"values":[["2010-02-28T01:03:37.703820946Z",200],["2012-02-28T01:03:38.703820946Z",300],["2015-02-28T01:03:36.703820946Z",100]]}]}]}`,
+		},
+		{
+			name:     "WHERE tags SELECT single field (NEQ tag value1 AND NEQ tag value2 AND NEQ tag value3)",
+			query:    `SELECT value FROM "%DB%"."%RP%".cpu WHERE host != 'server01' AND host != 'server02' AND host != 'server03'`,
+			expected: `{"results":[{}]}`,
+		},
+		{
+			reset: true,
```

Comment:
I added an explicit reset-and-write here, since this test is subtle. I thought it was worth making it explicit.